geyser-plugin-manager/src/geyser_plugin_manager.rs

```diff
@@ -14,13 +14,24 @@ use {
 pub struct LoadedGeyserPlugin {
     name: String,
     plugin: Box<dyn GeyserPlugin>,
+    // NOTE: While we do not access the library, the plugin we have loaded most
+    // certainly does. To ensure we don't SIGSEGV we must declare the library
+    // after the plugin so the plugin is dropped first.
+    //
+    // Furthermore, a well behaved Geyser plugin must ensure it ceases to run
+    // any code before returning from Drop. This means if the Geyser plugins
+    // spawn threads that access the Library, those threads must be `join`ed
+    // before the Geyser plugin returns from on_unload / Drop.
+    #[allow(dead_code)]
```

Comment:
The Library is loaded by the plugin manager -- not by the plugin itself. How is your plugin accessing the Library? Also, see https://github.com/anza-xyz/agave/blob/master/geyser-plugin-manager/src/geyser_plugin_manager.rs#L264 -- we were trying to ensure the plugin to be dropped before the library (the code). This is opposite what you are doing. Have you tested with dynamic plugin load/unload with your change? 
Can you also detail how to reproduce the issue?

Reply:
1. https://github.com/OliverNChalk/agave/commit/708e9a905483d9b7fb7f772c017877c4f9d14e0c
2. The struct drop order is top to bottom

Reply:
Struct fields are dropped in declaration order: https://doc.rust-lang.org/reference/destructors.html

This guarantees that the plugin is dropped before the library. The reverse was happening, which was the bug @OliverNChalk ran into.

As I could tell from code, the plugin manager was doing the correct drop order if plugin loaded successfully, the issue was only during the initial load. This approach of wrapping guarantees that the drop-order is correct in all places (after the struct is created).

Reply:
The change should be tested with dynamic unload/load though to make sure.

Reply:
> Can you also detail how to reproduce the issue?

PR description has steps to reproduce.
All that needs to happen is a plugin that returns Err from on_load.

Reply:
The two statements contradicts each other to me:

"This guarantees that the plugin is dropped before the library."

"When a plugin return an Err from on_load it will cause the plugin to be dropped before the library. This results in a SIGSEGV."

Reply:
`will cause the library to be dropped before the plugin`